CppSamples/Maps/ShowDeviceLocationUsingIndoorPositioning/ShowDeviceLocationUsingIndoorPositioning.cpp

```diff
@@ -79,11 +83,50 @@ void ShowDeviceLocationUsingIndoorPositioning::setMapView(MapQuickView* mapView)
   m_mapView = mapView;
   m_mapView->setMap(m_map);
 
-  setupIndoorsLocationDataSource();
+  #ifdef PERMISSIONS_PLATFORM
+    requestBluetoothThenLocationPermissions();
+  #else
+    setupIndoorsLocationDataSource();
+  #endif
 
   emit mapViewChanged();
 }
 
+void ShowDeviceLocationUsingIndoorPositioning::requestBluetoothThenLocationPermissions()
+{
+    qApp->requestPermission(QBluetoothPermission{}, [this](const QPermission& permission)
+  {
+    Q_UNUSED(permission);
+    requestLocationPermissionThenSetupILDS();
+  });
+}
+
+void ShowDeviceLocationUsingIndoorPositioning::requestLocationPermissionThenSetupILDS()
+{
+  QLocationPermission locationPermission{};
+  locationPermission.setAccuracy(QLocationPermission::Accuracy::Precise);
+  locationPermission.setAvailability(QLocationPermission::Availability::WhenInUse);
+  qApp->requestPermission(locationPermission, [this](const QPermission& permission)
+  {
+    Q_UNUSED(permission);
+    checkPermissions();
+    setupIndoorsLocationDataSource();
+  });
+}
+
+void ShowDeviceLocationUsingIndoorPositioning::checkPermissions()
+{
+  if (qApp->checkPermission(QBluetoothPermission{}) == Qt::PermissionStatus::Denied)
+  {
+    emit bluetoothPermissionDenied();
+  }
+
+  if (qApp->checkPermission(QLocationPermission{}) == Qt::PermissionStatus::Denied)
```

Comment:
How are the accuracy and availability parameters checked here? The `checkPermission` only checks the status right?

Reply:
If you don't explicitly state which types of `Accuracy` and `Availability` you are checking on the `QLocationPermission` passed into `checkPermission()`, you'll be checking the permission status of `Accuracy::Approximate` and `Availability::WhenInUse`. 

But you requested `Accuracy::Precise` and `Availability::WhenInUse`, because that is what is needed by ILDS.

Reply:
Added